src/App.jsx

```diff
@@ -14,33 +20,125 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getPrepearedGoods(goods, { sortField, isReversed }) {
+  const prepearedSortGoods = [...goods];
+
+  if (sortField) {
+    prepearedSortGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed === true) {
+    prepearedSortGoods.reverse();
+  }
+
+  return prepearedSortGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const sortGoods = getPrepearedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const isOriginalOrder =
+    JSON.stringify(sortGoods) === JSON.stringify(goodsFromServer);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_FIELD_ALPHABET,
+          })}
+          onClick={() => setSortField(SORT_FIELD_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SORT_FIELD_LENGTH,
+          })}
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': isReversed === false,
+          })}
+          onClick={() => setIsReversed(prev => !prev)}
+        >
+          Reverse
+        </button>
+
+        {!isOriginalOrder && (
+          <button
+            type="button"
+            className={cn('button', 'is-danger', {
+              'is-light': sortField !== SORT_FIELD_RESET,
+            })}
+            onClick={() => {
+              setSortField(SORT_FIELD_RESET);
+              setIsReversed(false);
+            }}
```

Comment:
Move this logic from jsx to the helper reset function and use it here
```suggestion
            onClick={reset}
```